src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -291,7 +342,37 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
         {
             assert(!db_and_table.first && !db_and_table.second);
             if (exception)
-                exception->emplace(Exception(ErrorCodes::UNKNOWN_TABLE, "Table {} doesn't exist", table_id.getNameForLogs()));
+            {
+                if (!db_and_table.first)
+                {
+                    DatabaseNameHints hints(*this);
+                    std::vector<String> names = hints.getHints(table_id.getDatabaseName(), hints.getAllRegisteredNames());
+                    /// I also leave possibility to print several suggestions
+                    if (!names.empty())
+                    {
+                    std::string suggested_name = names[0];
+                    exception->emplace(Exception(ErrorCodes::UNKNOWN_DATABASE, "Database {} doesn't exist. Maybe you wanted to type {}?", backQuoteIfNeed(table_id.getDatabaseName()), backQuoteIfNeed(suggested_name)));
```

Comment:
here and below
```suggestion
                    exception->emplace(Exception(ErrorCodes::UNKNOWN_DATABASE, "Database {} doesn't exist. Maybe you wanted to type {}?", backQuoteIfNeed(table_id.getDatabaseName()), backQuoteIfNeed(names[0])));
```

also we have `IHints::appendHintsMessage` which accepts hints array